src/core/pinned_memory_manager.cc

```diff
@@ -50,60 +51,102 @@ PointerToString(void* ptr)
 
 std::unique_ptr<PinnedMemoryManager> PinnedMemoryManager::instance_;
 
+class PinnedMemoryManagerImpl : public PinnedMemoryManager {
```

Comment:
Why did you make a Impl class instead of just directly implementing PinnedMemoryManager?

Reply:
There is no strong intention to do this. I was thinking that then a different implementation, if needed, can be added easily.